x-pack/plugins/apm/common/search_strategies/correlations/formatting_utils.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export function roundToDecimalPlace(
+  num?: number,
+  dp: number = 2
+): number | string {
+  if (num === undefined) return '';
+  if (num % 1 === 0) {
+    // no decimal place
+    return num;
+  }
+
+  if (Math.abs(num) < Math.pow(10, -dp)) {
+    return Number.parseFloat(String(num)).toExponential(2);
+  }
+  const m = Math.pow(10, dp);
+  return Math.round(num * m) / m;
+}
```

Comment:
It looks like everywhere numbers are formatted to 3 decimal places. In that case, can this be simplified to:

```suggestion
 export function asPreciseDecimal(value?: number | null) { 
   if (!isFiniteNumber(value)) { 
     return NOT_AVAILABLE_LABEL; 
   } 
  
   return numeral(value).format('0,0.000'); 
 } 
```

You can add it to:

https://github.com/elastic/kibana/blob/e580d5a1e2936bb9357d8ec5f12f7c50653937d8/x-pack/plugins/apm/common/utils/formatters/formatters.ts#L13-L19

Reply:
Updated here https://github.com/elastic/kibana/pull/99905/commits/31973bb263848694cde21111bc4f585f7e59ad7e